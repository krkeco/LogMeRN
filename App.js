import React, {Component} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

import {
  LineChart,
  BarChart,
  PieChart,
  ProgressChart,
  ContributionGraph,
  StackedBarChart
} from 'react-native-chart-kit'
 
import Navigator from './App/Navigator.js';

type Props = {};
export default class App extends Component<Props> {
 constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    return (
      <SafeAreaView>
        <Navigator/>
      </SafeAreaView>
    );
  }
}