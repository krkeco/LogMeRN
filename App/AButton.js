import React, {Component} from 'react';
import {
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  View,
  Text,
  Button,
  TextInput,
  StatusBar,
} from 'react-native';

export default class AButton extends Component<Props> {
 constructor(props) {
    super(props);
    this.state = {
    }
  }

  onPress = () => {
    if(this.props.onPress != undefined){
      this.props.onPress();
    }
  }

  render() {
    let color = '#5f9ea0';
    if(this.props.color != undefined){
      color = this.props.color;
    }
  
    return (
      <View 
      style={{ maxWidth: 150, height: 35, margin:5, padding:5, backgroundColor: color, borderRadius: 5}}>
         <TouchableOpacity 
          style={{}}
          onPress={() => this.onPress()}>
          <Text>{this.props.text}</Text>
        </TouchableOpacity>
      </View>
    );

  }
}