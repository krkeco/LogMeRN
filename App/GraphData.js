import React, {Component} from 'react';
import {
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  View,
  Text,
  Button,
  TextInput,
  Dimensions,
  StatusBar,
} from 'react-native';

import {
  LineChart,
  BarChart,
  PieChart,
  ProgressChart,
  ContributionGraph,
  StackedBarChart
} from 'react-native-chart-kit'


export default class GraphData extends Component<Props> {
 constructor(props) {
    super(props);
    this.state = {
      width: 0,
      logBook: [
      {
        data: [
          4,3,2,5,1
        ]
      },
      {
        data: [
          1,2,3,4,5
        ]
      }
      ],

    }
  }

  
  onLayout(e) {
    this.setState({width: Dimensions.get('window').width})
    // console.log(width, height)
  }


  onComponentWillMount(){
    // this.props.logBook.logs.map((log, index) => {
      
    // })
  }

  render() {

    return(
      <View
        onLayout={this.onLayout.bind(this)}>
        <Text>
          Bezier Line Chart {this.state.width}
        </Text>
        <LineChart
          data={{
            labels: ['January', 'February', 'March', 'April', 'May', 'June'],
            datasets: this.state.logBook
          }}
          width={this.state.width} // from react-native
          height={220}
          yAxisLabel={''}
          chartConfig={{
            backgroundColor: '#e26a00',
            backgroundGradientFrom: '#fb8c00',
            backgroundGradientTo: '#ffa726',
            decimalPlaces: 0, // optional, defaults to 2dp
            color: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
            style: {
              borderRadius: 16
            }
          }}
          bezier
          style={{
            marginVertical: 8,
            borderRadius: 16
          }}
        />
      </View>
    );

  }
}