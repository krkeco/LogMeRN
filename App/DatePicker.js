import React, {Component} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';

import {faCalendar} from '@fortawesome/free-solid-svg-icons'

import DateTimePicker from "react-native-modal-datetime-picker";

import AButton from './AButton.js';

type Props = {};

export default class DatePicker extends Component<Props> {
 constructor(props) {
    super(props);
    this.state = {
    }
  }


  showDateTimePicker = () => {
    this.setState({ isDateTimePickerVisible: true });
  };
 
  hideDateTimePicker = () => {
    this.setState({ isDateTimePickerVisible: false });
  };
 
  handleDatePicked = date => {
    console.log("A date has been picked: ", date);
    
    this.props.returnDate(date.getTime());
    
    this.hideDateTimePicker();
  };


  render() {
    let text = "";
    if(this.props.text != null){
      text = this.props.text;
    }

    return (
    <View>     
      <AButton 
      text={text}
      icon={faCalendar}
       onPress={this.showDateTimePicker} />
        <DateTimePicker
          isVisible={this.state.isDateTimePickerVisible}
          onConfirm={this.handleDatePicked}
          onCancel={this.hideDateTimePicker}
      />
    </View>
    );
  }
}