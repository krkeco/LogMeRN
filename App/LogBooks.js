import React, {Component} from 'react';
import {
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  View,
  Text,
  Button,
  TextInput,
  StatusBar,
} from 'react-native';

import DialogInput from 'react-native-dialog-input';
import AButton from './AButton.js';

export default class LogBooks extends Component<Props> {
 constructor(props) {
    super(props);
    this.state = {
      newLogBook: false,
      newLogBookText: 'logbook',
    }
  }

  addLogBook = (title) => {
    this.setState({newLogBook: false},this.props.saveData)
    this.props.books.push(this.props.newBook(title))
    
  }
  
  render() {
    let bookView = null;

    if(this.props.books != null){
        booksView = this.props.books.map((book,index) => {
          return <TouchableOpacity 
          key={index}
          onPress={() => this.props.selectBook(index)} >
            <Text style={{fontSize: 18, marginTop:10}}>book {book.name}</Text>
            {book.logs.map((log, index) => {
              return <Text key={index} style={{marginLeft: 10}}>{log.name}</Text>
            })}
          </TouchableOpacity>
        })
    }
    let addButton = <AButton
    onPress={()=>this.setState({newLogBook: true})} 
    text="Add a LogBook"
    />

    let newLogModal =
    <DialogInput isDialogVisible={this.state.newLogBook}
      title={"New LogBook"}
      message={"LogBook Title:"}
      hintInput ={"Your Logbook title"}
      submitInput={ (inputText) => {this.addLogBook(inputText)} }
      closeDialog={ () => {this.setState({newLogBook: false})}}>
    </DialogInput>

    return (
      <View>
        <Text style={{fontSize: 20}}>LogBooks:</Text>

        {booksView}

        {addButton}

        {newLogModal}

      </View>
    );

  }
}