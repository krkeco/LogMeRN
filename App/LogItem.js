import React, {Component} from 'react';
import {
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  View,
  Text,
  Button,
  TextInput,
  Modal,
  Picker,
  StatusBar,
} from 'react-native';

import AButton from './AButton.js';
import {getPrettyDate} from './Utils.js';



import { faAngleUp, faAngleDown, faPalette, faSave, faWindowClose } from '@fortawesome/free-solid-svg-icons'
import ColorPicker from './ColorPicker'

export default class LogItem extends Component<Props> {
 constructor(props) {
    super(props);
    this.state = {
      value: '',
      note: '',
      prettyDate: '',
      showData: false,
      isDateTimePickerVisible: false,
      viewColorModal: false,
      color: '#fff',
      floatColor: '#fff',
    }
  }


  componentDidMount(){
    // alert('lets check date')
   // this.checkInputData();
  }

  checkInputData = () => {
     Object.keys(this.props.log.values).map((value, index) =>{
      // alert('index'+index+' '+value+' '+this.props.prettyDate)
      if(value == this.props.prettyDate){
        // alert('this is a repeat date'+index+" "+JSON.stringify(this.props.log.values[value].value))
        this.setState({
          value: this.props.log.values[value].value, 
          note: this.props.log.values[value].note
        })
      }
    });
    this.setState({prettyDate: getPrettyDate(this.props.date)});
  }


  onColorChange = (color) =>{
    this.setState({ color: color })
    // alert(color)
  }

  setColor = () => {
    this.props.setColor(this.state.color);
  }

  render() {

    let colorModal = 
     <ColorPicker
     save={this.setColor}
     color={this.props.log.color}
     floatColor={this.state.color}
     onColorChange={this.onColorChange}
     viewColorModal={this.state.viewColorModal}
     toggle={()=>{this.setState({viewColorModal: !this.state.viewColorModal});}}
     />

  
    let holderValue = '';
    let holderNote = '';
    
    Object.keys(this.props.log.values).map((value, index) =>{
    
      if(value == this.props.prettyDate){
          holderValue = this.props.log.values[value].value;
          holderNote = this.props.log.values[value].note;
        }

    });

    
    let showDataView = <AButton
      color={this.props.log.color}
      icon={faAngleDown}
      onPress={() => this.setState({showData: !this.state.showData})}/>
    
    if(this.state.showData){
      showDataView = <View style={{alignItems:'center'}}>
      <AButton
      color={this.props.log.color}
      icon={faAngleUp}
      onPress={() => this.setState({showData: !this.state.showData})}/>

        {Object.keys(this.props.log.values).map((value, index) =>{
            
              return <TouchableOpacity 
                key={index} 
                onPress={() => {
                  let thisDate = new Date(value)//.getTime();
                  let newVal = (thisDate.getTime() + thisDate.getTimezoneOffset() * 60000);
                  // alert(thisDate+" "+value)
                  this.props.setDate(newVal);

                }}
                style={{ width: 300, backgroundColor: this.props.log.color, borderRadius: 5, margin: 5, alignItems:'center'}} >
                <Text  > {value} :: Value: {this.props.log.values[value].value}</Text>
                <Text> Note: {this.props.log.values[value].note}</Text>
                
              </TouchableOpacity>
        })}
      </View>
    }

    return (
      <View style={{flexDirection: 'column', backgroundColor: '#bbdefb', margin: 5, borderRadius: 5}} >
        <View >
          
          <View
            style={{height: 50, flex:1, flexDirection:'row'}}>
          
            
            <AButton
              color={this.props.log.color}
              onPress={() => {this.props.deleteLog(this.props.index)}}
              icon={faWindowClose}
              />
            
            {colorModal}
            
            <AButton
              icon={faPalette}
              color={this.props.log.color}
              onPress={()=>this.setState({viewColorModal: !this.state.viewColorModal})}/>
            
            <Text style={{height: 30, margin: 5, fontSize:25}} >{this.props.log.label}</Text>
            


              <Text style={{marginTop:22}}>log of: {this.props.prettyDate}</Text>

            </View>
            <View style={{flexDirection:'row',justifyContent: 'center',flex:1, alignItems: 'center'}}>
        
            
              <View>
              <Text>Value:</Text>      
               <TextInput
                placeholder={holderValue}
                style={{height: 35,width:50, borderColor: 'gray', borderWidth: 1, borderRadius: 5}}
                keyboardType="numeric"
                onChangeText={(text) => this.setState({value: text})}
                value={this.state.value}
              />
              </View>

              <View style={{marginLeft:10}}>
              <Text >Note/Text:</Text>
              <TextInput
                  placeholder={holderNote}
                  style={{height: 35,width: 200,  borderColor: 'gray', borderWidth: 1, borderRadius: 5}}
                  onChangeText={(note) => this.setState({note})}
                  
                  value={this.state.note}
                />
                </View>

              <View style={{paddingTop:20}}>
              <AButton            
                color={this.props.log.color}
                onPress={() => {this.props.saveLogData(this.props.date, this.state.value,this.state.note, this.props.log)}}
                icon={faSave}
                />
              </View>
              
              </View>

        </View>

          <View style={{alignItems: 'center'}}>
            {showDataView}
          </View>

      </View>
    );

  }
}