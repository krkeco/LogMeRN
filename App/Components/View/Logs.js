import React, { Component } from 'react';
import {
  StyleSheet,
  ScrollView,
  View,
  Text,
} from 'react-native';

import DialogInput from 'react-native-dialog-input';

import DatePicker from '../UI/DatePicker.js';
import Button from '../UI/Button.js';

import LogItem from '../Log/LogItem.js';

import GraphData from '../Graphing/GraphData.js';
import SampleLine from '../Graphing/SampleLine.js';

import { getPrettyDate } from '../../Utils.js';

import {
  faAngleLeft,
  faBars,
  faChartBar,
  faCalendar,
  faSave,
  faWindowClose,
  faPlus,
} from '@fortawesome/free-solid-svg-icons';

import {
  LineChart,
  BarChart,
  PieChart,
  ProgressChart,
  ContributionGraph,
  StackedBarChart,
} from 'react-native-chart-kit';

export default class Logs extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = {
      newLog: false,
      deleteLog: false,
      deleteBook: false,
      showGraph: false,
      newLogText: 'log',
      prettyDate: '2019-07-07',
    };
  }

  componentDidMount() {
    let d = new Date().getTime();
    this.setState({ prettyDate: getPrettyDate(d) });
  }

  addLog = (title) => {
    this.props.logBook.logs.push(this.newLog(title));

    this.setState({ newLog: false }, this.props.saveData);
  };

  newLog = (title) => {
    let r = 125 + Math.floor(Math.random() * 130);
    let g = 125 + Math.floor(Math.random() * 130);
    let b = 125 + Math.floor(Math.random() * 130);
    let o = 1;
    let colour = `rgba(${r}, ${g}, ${b}, ${o})`;

    let newLog = {
      color: colour,
      label: title,
      values: {},
    };
    return newLog;
  };

  logData = () => {
    alert('going to do something!');
  };

  saveLogData = (date, value, note, log) => {
    let key = date;
    log.values[key] = this.newLogObject(value, note);

    this.props.saveData();
  };
  newLogObject = (value, note) => {
    let newEntry = {
      value: value,
      note: note,
    };
    return newEntry;
  };

  returnDate = (date) => {
    this.setState({
      date: date,
      prettyDate: getPrettyDate(date),
    });
  };

  setColor = (color, log) => {
    log.color = color;
    this.props.saveData();
  };

  render() {
    let logs = null;
    if (this.props.logBook.logs != null) {
      logs = (
        <View>
          {this.props.logBook.logs.map((log, index) => {
            return (
              <LogItem
                setColor={(color) => this.setColor(color, log)}
                setDate={(date) => {
                  this.returnDate(date);
                }}
                prettyDate={this.state.prettyDate}
                saveLogData={(date, value, note, log) => {
                  this.saveLogData(date, value, note, log);
                }}
                deleteLog={(logIndex) => this.props.deleteLog(logIndex)}
                key={index}
                log={log}
                index={index}
              />
            );
          })}
        </View>
      );
    }

    let deleteButton = (
      <Button onPress={() => this.props.deleteBook()} icon={faWindowClose} />
    );

    let addButton = (
      <Button onPress={() => this.setState({ newLog: true })} icon={faPlus} />
    );

    let dateButton = <DatePicker returnDate={this.returnDate} />;

    let newLogModal = (
      <DialogInput
        isDialogVisible={this.state.newLog}
        title={'New Log'}
        message={'Log:'}
        hintInput={'Your Log'}
        submitInput={(inputText) => {
          this.addLog(inputText);
        }}
        closeDialog={() => {
          this.setState({ newLog: false });
        }}
      ></DialogInput>
    );

    let viewGraphButton = (
      <Button
        onPress={() => this.setState({ showGraph: true })}
        icon={faChartBar}
      />
    );

    let logBookMenu = (
      <View
        style={{
          flexDirection: 'row',
          flex: 1,
          justifyContent: 'space-between',
        }}
      >
        <View style={{ flexDirection: 'row' }}>
          {addButton}

          {newLogModal}

          {dateButton}
        </View>

        {viewGraphButton}
      </View>
    );

    if (this.props.logBook.logs.length < 1) {
      logBookMenu = (
        <View style={{ flexDirection: 'row', flex: 1 }}>
          {addButton}

          {newLogModal}
        </View>
      );
    }
    if (this.state.showGraph) {
      logBookMenu = null;

      logs = (
        <SampleLine
          goBack={() => this.setState({ showGraph: false })}
          logBook={this.props.logBook}
        />
      );
    }

    return (
    <View style={{height: '100%'}} >
        <View style={{ flexDirection: 'row' }}>
          <Button icon={faAngleLeft} onPress={this.props.goBack} />

          <Text style={{ fontSize: 32, flex: 1, marginLeft: 5 }}>
            {this.props.logBook.name}
          </Text>
          {deleteButton}
        </View>
        <View style={{flexDirection:'row'}}>
        {logBookMenu}
        </View>
      <ScrollView>
        {logs}
      </ScrollView>
      </View>
    );
  }
}
