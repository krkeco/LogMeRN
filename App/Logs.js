import React, {Component} from 'react';
import {
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  View,
  Button,
  Picker,
  TextInput,
  Text,
  StatusBar,
  Alert,
} from 'react-native';

// import Storage from 'react-native-storage';
// import AsyncStorage from '@react-native-community/async-storage';

import DatePicker from './DatePicker.js';
import DialogInput from 'react-native-dialog-input';
import AButton from './AButton.js';
import LogItem from './LogItem.js';
import GraphData from './GraphData.js';
import SampleLine from './SampleLine.js';

 
import {getPrettyDate} from './Utils.js';

import { faAngleLeft, faBars, faChartBar, faCalendar, faSave, faWindowClose, faPlus } from '@fortawesome/free-solid-svg-icons'


import {
  LineChart,
  BarChart,
  PieChart,
  ProgressChart,
  ContributionGraph,
  StackedBarChart
} from 'react-native-chart-kit'

export default class Logs extends Component<Props> {
 constructor(props) {
    super(props);
    this.state = {
      newLog: false,
      deleteLog: false,
      deleteBook: false,
      showGraph: false,
      newLogText: 'log',
      date: 1562482800000,
      prettyDate: '2019-07-07',
    }
  }

  componentDidMount(){
    
  }

  addLog = (title) => {
    this.props.logBook.logs.push(this.newLog(title));
    
    this.setState({newLog: false},this.props.saveData);
  }

  newLog = (title) =>{
    let r = 125+ Math.floor(Math.random()*130);
      let g = 125+ Math.floor(Math.random()*130);
      let b = 125+ Math.floor(Math.random()*130);
      let o = 1;
      let colour = `rgba(${r}, ${g}, ${b}, ${o})`;

    let newLog = {
      "color":colour,
      "label":title,
      "values":{}
      // "values":[]
    }
    return newLog;
  }

  logData = () => {
    alert('going to do something!')
  }

  saveLogData = (date,value,note,log) => {

    let key = getPrettyDate(date);
    log.values[key] = this.newLogObject(value,note);
    
    this.props.saveData();

  }
  // newLogEntry =  (value,date,note) => {
  //   let newEntry = {
  //     "date": date,
  //     "value": value,
  //     "note": note,
  //   }
  //   return newEntry;
  // } 
  newLogObject =  (value,note) => {
    let newEntry = {
      "value": value,
      "note": note,
    }
    return newEntry;
  }

  returnDate = (date) => {
    this.setState({
      date: date,
      prettyDate: getPrettyDate(date)});

  }

  setColor = (color,log) => {
    log.color = color;
    this.props.saveData();
  }

  render() {
    // let logs = null;
    if(this.props.logBook.logs != null){
        logs = <View >
        {this.props.logBook.logs.map((log,index) => {
              return <LogItem
                setColor={(color) => this.setColor(color,log)}
                setDate={(date)=>{this.returnDate(date)}}
                date={this.state.date}
                prettyDate={this.state.prettyDate}
                saveLogData={(date,value,note,log) => {this.saveLogData(date,value,note,log)}}
                deleteLog={(logIndex) => this.props.deleteLog(logIndex)}
                key={index}
                log={log}
                index={index}/>
        })}
        </View>
    }

    let deleteButton = <AButton
      onPress={() => this.props.deleteBook()} 
      icon={faWindowClose}
      />

    let addButton = <AButton
      onPress={() => this.setState({newLog: true})} 
      icon={faPlus}
      />

      let dateButton = <DatePicker
        returnDate={this.returnDate}/>

    let viewGraphButton = <AButton
      onPress={() => this.setState({showGraph: true})}
      icon={faChartBar}/>

    if(this.state.showGraph){
      viewGraphButton = null;

      logs = <SampleLine
        goBack={()=>this.setState({showGraph:false})}
        logBook={this.props.logBook}/>
      // <GraphData
      //   goBack={()=>this.setState({showGraph: false})}
      //   logBook={this.props.logBook}/>;

    }


    let newLogModal =
    <DialogInput isDialogVisible={this.state.newLog}
      title={"New Log"}
      message={"Log:"}
      hintInput ={"Your Log"}
      submitInput={ (inputText) => {this.addLog(inputText)} }
      closeDialog={ () => {this.setState({newLog: false})}}>
    </DialogInput>;

   


    let logBookMenu =
      <View
        style={{flexDirection:'row',flex:1, justifyContent:'space-between'}}>
        
        <View style={{flexDirection:'row'}}>
          {addButton}

          {newLogModal}

          {dateButton}
        </View>

        {viewGraphButton}
      
      </View>;

    if(this.props.logBook.logs.length < 1){
      logBookMenu =
            <View
        style={{flexDirection:'row',flex:1}}>
        
        {addButton}

        {newLogModal}
      
      </View>;
    }

    return (
      <ScrollView>
        <View style={{flexDirection:'row'}}>
          <AButton
            icon={faAngleLeft}
            onPress={this.props.goBack}/>

          <Text style={{fontSize: 32, flex:1, marginLeft: 5}} >{this.props.logBook.name}</Text>
          {deleteButton}
        </View>
        
        {logBookMenu}


        {logs}
      
      </ScrollView>
    );

  }
}