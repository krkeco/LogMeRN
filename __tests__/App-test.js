/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import AButton from '../App/AButton'
import LogItem from '../App/LogItem'
import Logs from '../App/Logs'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
jest.mock('@fortawesome/react-native-fontawesome', () => 'FontAwesomeIcon');
// import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
// it('renders correctly', () => {
//   renderer.create(<App />);
// });


//sample data entries and test it
let data = {"2019-07-07":{"value":"5","note":"note"},"2019-07-09":{"value":"9","note":"not a note"}}

test('entry values', () => {
  expect(data["2019-07-07"].value).toBe("5");
});
test('entry note', () => {
  expect(data["2019-07-09"].note).toBe("not a note");
});

//sample log and test it
let log = {"label":"patience","values":data,"color":"#fff"}

test('log entry values', () => {
  expect(log.values["2019-07-07"].value).toBe("5");
});
test('log color', () => {
  expect(log.color).toBe("#fff");
});

let newLog = Logs.newLog('sampleLog');
test('log color', () => {
  expect(newLog.label).toBe("sampleLog");
});


let ledger = {
  "name":"birthdays",
  logs: [
    log
  ]
}

test('ledger entry values', () => {
  expect(ledger.name).toBe("birthdays");
});

it('abutton renders correctly', () => {
  const tree = renderer.create(
    <AButton />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

// it('renders correctly', () => {
//   const tree = renderer.create(
//     <LogItem 
//     	log={log}
//     	setColor={null}
//         setDate={null}
//         prettyDate="2019-07-07"
//         saveLogData={null}
//         deleteLog={null}
//         key={1}
//         index={1}/>
//     ).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// it('renders correctly', () => {
//   const tree = renderer.create(
//     <AButton />
//     ).toJSON();
//   expect(tree).toMatchSnapshot();
// });
// it('renders correctly', () => {
//   const tree = renderer.create(
//     <AButton />
//     ).toJSON();
//   expect(tree).toMatchSnapshot();
// });