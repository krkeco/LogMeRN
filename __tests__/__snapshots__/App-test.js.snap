// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`abutton renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#bbdefb",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 35,
        "margin": 5,
        "maxWidth": 150,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#bbdefb",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 35,
        "margin": 5,
        "maxWidth": 150,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#bbdefb",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 35,
        "margin": 5,
        "maxWidth": 150,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#bbdefb",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 35,
        "margin": 5,
        "maxWidth": 150,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
  </View>
</View>
`;
